tests/checkasm/vvc_deblock.c

```diff
@@ -72,59 +72,96 @@ static void randomize_params(int32_t beta[4], int32_t tc[4], const int size)
     }
 }
 
-static void randomize_chroma_buffers(int type, int beta[4], int32_t tc[4],
-   uint8_t *buf, ptrdiff_t xstride, ptrdiff_t ystride, const int shift, const int bit_depth)
+static void randomize_chroma_buffers(const int bit_depth, const int type, const int shift, const int spatial_strong,
+uint8_t* max_len_p, uint8_t* max_len_q, int beta[4], int32_t tc[4], 
+uint8_t *buf0, uint8_t *buf1, ptrdiff_t xstride, ptrdiff_t ystride)
 {
     const int size = shift ? 4 : 2;
     const int end = 8 / size;
 
+    switch (type) {
+        case 0: // strong
+            for(int i = 0; i < 4; ++i) {
+                max_len_p[i] = 3;
+                max_len_q[i] = 3;
+            }
+            break;
+        case 1: // strong one-side
+            for(int i = 0; i < 4; ++i) {
+                max_len_p[i] = 1;
+                max_len_q[i] = 3;
+            }
+            break;
+        case 2: // weak
+            for(int i = 0; i < 4; ++i) {
+                max_len_p[i] = 1;
+                max_len_q[i] = 1;
+            }
+            break;
+        case 3: // mix
+            for(int i = 0; i < 4; ++i) {
+                max_len_p[i] = 1 + (rnd() % 2) * 2;
+                max_len_q[i] = 1 + (rnd() % 2) * 2;
+                max_len_p[i] = FFMIN(max_len_p[i], max_len_q[i]);
+            }
+            break;
+    }
+
     randomize_params(beta, tc, size);
-    for (int j = 0; j < size; j++)
-    {
-        const int tc25 = TC25(j) << (bit_depth - 10);
-
-        const int tc25diff = FFMAX(tc25 - 1, 0);
-        // 2 or 4 lines per tc
-        for (int i = 0; i < end; i++)
-        {
-            int b3diff;
-            int b3 = (beta[j] << (bit_depth - 8)) >> 3;
-
-            SET(P0, rnd() % (1 << bit_depth));
-            SET(Q0, RANDCLIP(P0, tc25diff));
-
-            // p3 - p0 up to beta3 budget
-            b3diff = rnd() % b3;
-            SET(P3, RANDCLIP(P0, b3diff));
-            // q3 - q0, reduced budget
-            b3diff = rnd() % FFMAX(b3 - b3diff, 1);
-            SET(Q3, RANDCLIP(Q0, b3diff));
-
-            // same concept, budget across 4 pixels
-            b3 -= b3diff = rnd() % FFMAX(b3, 1);
-            SET(P2, RANDCLIP(P0, b3diff));
-            b3 -= b3diff = rnd() % FFMAX(b3, 1);
-            SET(Q2, RANDCLIP(Q0, b3diff));
-
-            // extra reduced budget for weighted pixels
-            b3 -= b3diff = rnd() % FFMAX(b3 - (1 << (bit_depth - 8)), 1);
-            SET(P1, RANDCLIP(P0, b3diff));
-            b3 -= b3diff = rnd() % FFMAX(b3 - (1 << (bit_depth - 8)), 1);
-            SET(Q1, RANDCLIP(Q0, b3diff));
-
-            buf += ystride;
+    randomize_buffers(buf0, buf1, BUF_SIZE);
+
+    uint8_t* buf = buf0 + xstride * 5;
+
+    if (spatial_strong) {
+        for (int j = 0; j < size; j++) {
+            const int tc25 = TC25(j) << (bit_depth - 10);
```

Comment:
weak, strong, one side strong can be mixed.
maybe we need to add some commentslike hevc_deblock.c